geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/SetClusterConfigurationCommand.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.cli.result.model.TabularResultModel;
+import org.apache.geode.management.internal.configuration.domain.Configuration;
+import org.apache.geode.management.internal.configuration.functions.GetRegionNamesFunction;
+import org.apache.geode.management.internal.configuration.functions.RecreateCacheFunction;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+
+public class SetClusterConfigurationCommand extends InternalGfshCommand {
+  public static final String COMMAND = "set-cluster-configuration";
+  public static final String COMMAND_HELP =
+      "this commands overwrites the existing cluster configuration for the specified group.";
+  public static final String XML_FILE = "xml-file";
+  public static final String CONFIGURE_RUNNING_SERVERS = "configure-running-servers";
+  public static final String CONFIGURE_HELP =
+      "Running servers will be configured with the content of xml file only when the these servers do not have any existing configurations.";
+  public static final String IGNORE_RUNNING_SERVERS = "ignore-running-servers";
+  public static final String IGNORE_HELP =
+      "Running servers will be left alone. It's assumed that the content of the xml you are setting here has no conflict with these servers' configuration.";
+
+
+  @CliCommand(value = COMMAND, help = COMMAND_HELP)
+  @CliMetaData(
+      interceptor = "org.apache.geode.management.internal.cli.commands.SetClusterConfigurationCommand$Interceptor",
+      relatedTopic = {CliStrings.TOPIC_GEODE_CONFIG})
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public ResultModel setConfig(
+      @CliOption(key = CliStrings.GROUP,
+          unspecifiedDefaultValue = ConfigurationPersistenceService.CLUSTER_CONFIG) String group,
+      @CliOption(key = XML_FILE, mandatory = true) String xmlFile,
+      @CliOption(key = CONFIGURE_RUNNING_SERVERS, help = CONFIGURE_HELP,
+          specifiedDefaultValue = "true", unspecifiedDefaultValue = "false") boolean configure,
+      @CliOption(key = IGNORE_RUNNING_SERVERS, help = IGNORE_HELP, specifiedDefaultValue = "true",
+          unspecifiedDefaultValue = "false") boolean ignore)
+      throws IOException, TransformerException, ParserConfigurationException, SAXException {
+    if (!isSharedConfigurationRunning()) {
+      return ResultModel.createError("Cluster configuration service is not running.");
+    }
+
+    InternalConfigurationPersistenceService ccService = getConfigurationPersistenceService();
+    Set<DistributedMember> serversInGroup = findMembers(group);
+
+    ResultModel result = new ResultModel();
+    InfoResultModel infoSection = result.addInfo(ResultModel.INFO_SECTION);
+    ccService.lockSharedConfiguration();
+    try {
```

Comment:
Would it be cleaner if we introduce (in another ticket) a `try`-with-resources lock for the configuration service?  If we had the configuration service implement `AutoClosable` to release the lock and a method `getAndLockConfigurationPersistenceService`, it could tidy up the declaration and enforce the `finally` block.

Reply:
a good idea.